code/game/objects/structures/vehicles/vehicle.dm

```diff
@@ -95,16 +96,56 @@
 			to_chat(user, "Need more welding fuel!")
 			return
 	else if(istype(W, /obj/item/key))
-		if(keytype)
-			to_chat(user, "Hold \the [W] in one of your hands while you drive \the [src].")
+		if(!heldkey)
+			if(keytype && mykey == W)
+				if(prob(95) && user.drop_item(W, src))
+					to_chat(user, "<span class='notice'>You insert \the [W] to \the [src]'s ignition and turn it.</span>")
+					user.visible_message("<span class='notice'>\The [src]'s engine roars to life!</span>")
+					playsound(get_turf(src), "sound/items/screwdriver.ogg", 10, 1) // find a better sound later
+					src.heldkey = W
+				else
+					to_chat(user, "<span class='warning'>You try to insert \the [W] to \the [src]'s ignition but you miss the slot!</span>")
+			else
+				if(keytype)
+					to_chat(user, "<span class='warning'>\The [W] doesn't fit into \the [src]'s ignition.</span>")
+				else
+					to_chat(user, "<span class='notice'>You don't need a key.</span>")
 		else
-			to_chat(user, "You don't need a key.")
+			to_chat(user, "<span class='notice'>\the [src]'s already has \the [heldkey] in it.</span>")
```

Comment:
1. Should be `\The` at the beginning
2. Looks like you either forgot to add a word or forgot to remove `'s`